taqueria-plugin-taquito/originate.ts

```diff
@@ -17,185 +24,167 @@ interface OriginationResult {
-        : config.network[networkName]
+            return [
+                ...retval, {
+                    contract: contract.filename,
+                    address: "Error",
+                    destination
+                }
+            ]
+        },
+        [] as OriginationResult[]
+    )
 }
 
-const getSandboxConfig = (sandboxName: string, config: Config) => {
-    return !config.sandbox[sandboxName]
-    ? undefined
-    : config.sandbox[sandboxName]
-}
 
-const getAccountSecretKey = (sandbox: SandboxConfig) => {
-    if (sandbox.accounts && sandbox.accounts.default) {
-        const accountName = (sandbox.accounts.default as string);
-        const accountDetails = sandbox.accounts[accountName] as AccountDetails
-        if (accountDetails.keys) return accountDetails.keys.secretKey.replace(/unencrypted:/, '')
+const createBatch = async (parsedArgs: Opts, tezos: TezosToolkit, destination: string) => {
+    const contracts = await getValidContracts (parsedArgs)                                
+    
+    const batch = await contracts.reduce(
+        (batch, contractMapping) => 
+            contractMapping.storage
+                ? addOrigination(parsedArgs, batch) (contractMapping)
+                : batch,
+        Promise.resolve(tezos.wallet.batch())
+    )
+    
+    try {
+        const op = await batch.send()
+        const confirmed = await op.confirmation()
+        return await mapOpToContract(contracts, op, destination)
+    }
+    catch (err) {
+        const error = (err as {message: string})
+        if (error.message) sendErr(error.message)
+        return undefined
     }
-
-    return undefined        
 }
 
-const originateContract = (parsedArgs: Opts) => (contractFilename: string) => {
-    // TODO: Should getting the default environment be provided by the SDK or the framework?
-    const currentEnv = parsedArgs.env
-        ? (parsedArgs.env as string)
-        : (
-            parsedArgs.config.environment
-                ? parsedArgs.config.environment.default
-                : 'development'
+const originateToNetworks = (parsedArgs: Opts, currentEnv: EnvironmentConfig) =>
+    currentEnv.networks
+        ? currentEnv.networks.reduce(
+            (retval, networkName) => {
+                const network = getNetworkConfig (parsedArgs) (networkName)
+                if (network.rpcUrl) {
+                    if (network.faucet) {
+                        const result = (async () => {
+                            const tezos = new TezosToolkit(network.rpcUrl as string)
+                            await importKey(tezos, network.faucet.email, network.faucet.password, network.faucet.mnemonic.join(' '), network.faucet.activation_code)
+                            return await createBatch(parsedArgs, tezos, networkName)
+                        })()
+
+                        return [...retval, result]
+                    }
+                    else sendErr(`Network ${networkName} requires a valid faucet`)
```

Comment:
For best ux, could we point the user to where they fix it with something like:
```
Network ${networkName} requires a valid faucet to be configured in `config.json`
```?

Reply:
Good idea. I've made that adjustment.